root/frontend/src/components/Weather/Weather.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import useStyles from './Weather.style';
+
+function Weather() {
+  const classes = useStyles();
+  const fetchWeatherData = async (): Promise<any> => {
+    const data = await axios('http://localhost:5005/weather/getWeatherInfo');
+    return data.data;
+  };
+
+  type WeatherDataObj = {
+    current: any;
+    location: any;
+  };
+
+  const [weatherData, setWeatherData] = useState<WeatherDataObj>();
+
+  useEffect(() => {
+    const getWeather = async () => {
+      const fetchedWeatherData = await fetchWeatherData();
+      setWeatherData(fetchedWeatherData);
```

Comment:
Same deal here as listed above with `fetchWeatherData()`. TL;DR: this function isn't listed in the dependencies of the `useEffect()`.